src/core/genfb.h

```diff
@@ -156,24 +145,77 @@ class CGenFunctionBlock : public T {
     static size_t getDataPointSpecSize(const CIEC_ANY &paValue);
     static void fillDataPointSpec(const CIEC_ANY &paValue, CStringDictionary::TStringId *&paDataTypeIds);
 
-    static size_t calculateFBConnDataSize(const SFBInterfaceSpec &paInterfaceSpec);
+    virtual void setupFBInterface();
 
-    static size_t calculateFBVarsDataSize(const SFBInterfaceSpec &paInterfaceSpec);
+    virtual void createGenInputData() {
+      // TODO turn into pure virtual as soon as all gen fbs are converted
```

Comment:
maybe it's fine as it is now, since not all subclasses have intput AND output generic data, right? Otherwise you'll also spread a lot of empty implementations in all generic FBs